conans/client/remote_manager.py

```diff
@@ -199,10 +199,14 @@ def search_recipes(self, remote, pattern, ignorecase=True):
 
         return self._call_remote(remote, "search", pattern)
 
-    def search_packages(self, remote, ref, query):
-        packages = self._call_remote(remote, "search_packages", ref, query)
-        packages = filter_packages(query, packages)
-        return packages
+    def search_packages(self, remote, ref):
+        infos = self._call_remote(remote, "search_packages", ref)
+        ret = OrderedDict()
+        for package_id, data in infos.items():
+            # FIXME: we don't have the package reference, it uses the latest, we could check
```

Comment:
Lets try to improve server side if possible.

Reply:
Already added to the doc of Artifactory improvements.